scripts/reborn_qa_matrix/run_hermetic_qa.py

```
@@ -0,0 +1,678 @@
+    return names
+
+
+def _test_ids_for(cases: list[CaseSpec]) -> list[str]:
+    return sorted({test_id for case in cases for test_id in case.qa_matrix_test_ids})
+
+
+def _case_has_matrix_only_command(case: CaseSpec) -> bool:
+    return bool(case.commands)
+
+
+def _case_existing_ci_only(case: CaseSpec) -> bool:
```

Ruff (0.15.20)

[warning] Unused function argument: `case`

(ARG001)